hack/tools/fetch-images/main.go

```
@@ -195,21 +221,50 @@ func getHelmValues(carenChartDirectory string) (map[string]interface{}, error) {
 	return m, nil
 }
 
-func getValuesFileForChartIfNeeded(chartName, carenChartDirectory string) string {
+func getValuesFileForChartIfNeeded(chartName, carenChartDirectory string) (string, error) {
 	switch chartName {
 	case "nutanix-csi-storage":
-		return path.Join(carenChartDirectory, "addons", "csi", "nutanix", defaultHelmAddonFilename)
+		return filepath.Join(carenChartDirectory, "addons", "csi", "nutanix", defaultHelmAddonFilename), nil
 	case "node-feature-discovery":
-		return path.Join(carenChartDirectory, "addons", "nfd", defaultHelmAddonFilename)
+		return filepath.Join(carenChartDirectory, "addons", "nfd", defaultHelmAddonFilename), nil
 	case "snapshot-controller":
-		return path.Join(carenChartDirectory, "addons", "csi", "snapshot-controller", defaultHelmAddonFilename)
+		return filepath.Join(carenChartDirectory, "addons", "csi", "snapshot-controller", defaultHelmAddonFilename), nil
 	case "cilium":
-		return path.Join(carenChartDirectory, "addons", "cni", "cilium", defaultHelmAddonFilename)
+		return filepath.Join(carenChartDirectory, "addons", "cni", "cilium", defaultHelmAddonFilename), nil
+	// Calico values differ slightly per provider, but that does not have a material imapct on the images required
+	// so we can use the default values file for AWS provider.
+	case "tigera-operator":
+		f := filepath.Join(carenChartDirectory, "addons", "cni", "calico", "aws", defaultHelmAddonFilename)
+		tempFile, err := os.CreateTemp("", "")
+		if err != nil {
+			return "", fmt.Errorf("failed to create temp file: %w", err)
+		}
+
+		// CAAPH uses unstructured internally, so we need to create an unstructured copy of a cluster
+		// to pass to the CAAPH values template.
+		c, _ := runtime.DefaultUnstructuredConverter.ToUnstructured(&clusterv1.Cluster{})
```

we seem to be ignoring an error here

---

Added error check.